### PR TITLE
feat: API Key to Access Token

### PR DESCRIPTION
**Pull request by Bento007**
### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add curator_api.yml
- Use blueprints to import connexion apis
- Use connexion to display openapi spec.
- Add POST curator/auth/token to retrieve an auth token for the curator API.
- add unit tests and smoke tests for curator auth token API.

- Change the path to the dataportal openapi spec to /
- add the path to the curator openapi spec to /curator/

## Definition of Done (from ticket)
#2012 
#2011

### CODE REVIEW

File: backend/corpora/api_server/app.py [hunk 1]
```diff
@@ -19,8 +19,14 @@ def create_flask_app():
     connexion_app = connexion.FlaskApp(f"{APP_NAME}-{DEPLOYMENT_STAGE}", specification_dir="backend/config")
     # From https://github.com/zalando/connexion/issues/346
     connexion_app.app.url_map.strict_slashes = False
-    swagger_spec_path = f"{APP_NAME}.yml"
-    connexion_app.add_api(swagger_spec_path, validate_responses=True)
+    dataportal_api = super(connexion.FlaskApp, connexion_app).add_api(
```
Comment (danieljhegeman) [inline backend/corpora/api_server/app.py:22]: why is this `super` call necessary?

File: backend/corpora/api_server/app.py [hunk 2]
```diff
@@ -19,8 +19,14 @@ def create_flask_app():
...
+        f"{APP_NAME}.yml", validate_responses=True, options={"swagger_ui": True}
+    )
+    curator_api = super(connexion.FlaskApp, connexion_app).add_api(
+        "curator-api.yml", validate_responses=True, options={"swagger_ui": True}
```
Comment (danieljhegeman) [inline backend/corpora/api_server/app.py:26]: I'm noticing that we didn't specify `swagger_ui: True` previously. It existed by default though, right? But we aren't publicly advertising the url...? 🤔

Comment (Bento007) [reply backend/corpora/api_server/app.py:26]: Yes it wasn't advertised. I'm being explicit here for simplicity.

File: backend/corpora/api_server/app.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 import json
 import logging
 import os
-from connexion import FlaskApi, ProblemException, problem
-from flask import g, Response, jsonify
+from connexion import FlaskApi, ProblemException, problem, Api
```
Comment (metakuni) [inline backend/corpora/api_server/app.py:6]: `Api` is an unused import, claims lint.

**Review by metakuni — APPROVED** [review]
What is the empty file `backend/config/create_collection copy.txt` for? 🤔

File: tests/unit/backend/corpora/api_server/test_app.py [hunk 4]
```diff
@@ -2,7 +2,11 @@
...
 class TestAPI(BaseAPITest):
-    def test_smoke(self):
+    def test_smoke_dataportal(self):
         """If this fails then the server does not work"""
-        response = self.app.get("/")
+        response = self.app.get("/ui/")
+        self.assertEqual(200, response.status_code)
+
+    def test_smoke_curator(self):
+        response = self.app.get("/curation/ui/")
```
Comment (danieljhegeman) [inline tests/unit/backend/corpora/api_server/test_app.py:11]: `ui`?

File: backend/corpora/api_server/app.py (hunk 1 above)
Comment (danieljhegeman) [reply backend/corpora/api_server/app.py:23]: what about [this comment](https://github.com/spec-first/connexion/issues/512#issuecomment-570658527) on that issue's thread?

Comment (danieljhegeman) [reply backend/corpora/api_server/app.py:23]: It also takes care of the blueprint registration below, it seems. Correct me if I'm reading it wrong.

File: backend/corpora/api_server/app.py [hunk 5]
```diff
@@ -19,8 +19,15 @@ def create_flask_app():
...
     # From https://github.com/zalando/connexion/issues/346
     connexion_app.app.url_map.strict_slashes = False
-    swagger_spec_path = f"{APP_NAME}.yml"
-    connexion_app.add_api(swagger_spec_path, validate_responses=True)
+    options = {"swagger_ui": True, 'swagger_url': '/'}
+    dataportal_api = super(connexion.FlaskApp, connexion_app).add_api(
+        f"{APP_NAME}.yml", validate_responses=True, options=options
+    )
+    curator_api = super(connexion.FlaskApp, connexion_app).add_api(
+        "curation-api.yml", validate_responses=True, options=options
```
Comment (danieljhegeman) [inline backend/corpora/api_server/app.py:27]: is there any reason to have the app come in as an env variable `APP_NAME` but not for the curation api? Should we hardcode both? 🤔

File: backend/corpora/api_server/app.py (hunk 1 above)
Comment (danieljhegeman) [reply backend/corpora/api_server/app.py:23]: Oh, but we have to have be able to specify the `url_prefix` when we make the call to `register_blueprint`. Nvm then.

**Review by danieljhegeman — APPROVED** [review]

**Pull request merged**